src/Microsoft.DotNet.Wpf/src/WindowsBase/System/Windows/Interop/ComponentDispatcherThread.cs

```diff
@@ -170,12 +170,12 @@ public void RemoveThreadPreprocessMessageHandlerFirst(ThreadMessageEventHandler
             {
                 ThreadMessageEventHandler newHandler = null;
 
-                foreach (ThreadMessageEventHandler testHandler in _threadPreprocessMessage.GetInvocationList())
+                foreach (ThreadMessageEventHandler testHandler in Delegate.EnumerateInvocationList(_threadPreprocessMessage))
```

Comment:
I'd remove the outer null checks as `EnumerateInvocationList` accepts null and returns an empty enumerator (false). Same in `PropertyMetadata`. Otherwise LGTM.

Reply:
I'd rather keep it that way since it would make cases when _threadPreprocessMessage is null slower while maybe making a tiny perf gain by skipping a null check. Your suggestion would require knowing how often it is called with a null delegate vs a non null delegate, which is more complex.